components/opponent/wikis/warcraft/opponent_display_custom.lua

```diff
@@ -25,7 +25,7 @@ CustomOpponentDisplay.propTypes.InlineOpponent = TypeUtil.extendStruct(OpponentD
 	showRace = 'boolean?',
 })
 
----@class WarcraftInlineOpponentProps
+---@class WarcraftInlineOpponentProps: InlineOpponentProps
```

Comment:
tldr: it stops some annotation errors

langer version:
- basically we are misusing classes from time to time to define bigger tables with known keys
- this extends such a class so that it knows we basically have a standard opponent properties class but with some additions for warcraft/starcraft

Reply:
It's inheritance, yeah
The class `WarcraftInlineOpponentProps` is a child class to the class `InlineOpponentProps`.

As children can obviously be provided when their abstract class/interface "parent" is expected.